qadence/ml_tools/config.py

```diff
@@ -70,3 +81,265 @@ def __post_init__(self) -> None:
+                        will be used."
+                    )
+
+                    self.feature_map_strategy = "analog"
+
+            elif self.feature_map_strategy == "analog":
+                if isinstance(self.operation, ParametricBlock):
+                    logger.warning(
+                        "The `operation` is a digital gate but the `feature_map_strategy` is\
+                        `analog`. The `feature_map_strategy` will be modified and given operation\
+                        will be used."
+                    )
+
+                    self.feature_map_strategy = "digital"
+
+        if self.inputs is not None:
+            assert (
+                len(self.inputs) == self.num_features
+            ), "Inputs list must be of same size as the number of features"
+        else:
+            if self.num_features == 1:
+                self.inputs = ["x"]
+            else:
+                raise ValueError(
+                    """
+                    Your QNN has more than one input. Please provide a list of inputs in the order
+                    of your tensor domain. For example, if you want to pass
+                    `xs = torch.rand(batch_size, input_size:=3)` to you QNN, where
+                    ```
+                    t = x[:,0]
+                    x = x[:,1]
+                    y = x[:,2]
+                    ```
+                    you have to specify
+                    ```
+                    inputs=["t", "x", "y"]
+                    ```
+                    You can also pass a list of sympy symbols.
+                """
+                )
+
+        property_list = [
+            "basis_set",
+            "reupload_scaling",
+            "feature_range",
+            "target_range",
+            "num_repeats",
+        ]
+
+        for target_field in fields(self):
+            if target_field.name in property_list:
+                prop = getattr(self, target_field.name)
+                if isinstance(prop, dict):
+                    assert set(prop.keys()) == set(
+                        self.inputs
+                    ), f"The keys in {target_field.name} must be the same as the inputs provided. \
+                    Alternatively, provide a single value of {target_field.name} to use the same\
+                    {target_field.name} for all features."
+                else:
+                    prop = {key: prop for key in self.inputs}
+                    setattr(self, target_field.name, prop)
+
+
+@dataclass
+class AnsatzConfig:
+    depth: int = 1
+    """Number of layers of the ansatz."""
+
+    ansatz_type: str = "hea"
+    """What type of ansatz.
+
+    "hea" for Hardware Efficient Ansatz.
+    "iia" for Identity intialized Ansatz.
+    """
+
+    ansatz_strategy: str = "digital"
+    """Ansatz strategy.
+
+    "digital" for fully digital ansatz. Required if `ansatz_type` is `iia`.
+    "sdaqc" for analog entangling block.
+    "rydberg" for fully rydberg hea ansatz.
```

Comment:
Maybe you should just add Rydberg to the `Strategy` `StrEnum` and use that everywhere that you have "digital", "sdaqc", "analog".

Reply:
On it.